### PR TITLE
refactor(fs): prefer our fs and fs to fs-extra (pt1: non-test files)

### PR DESCRIPTION
**Pull request by Hweinstock**
## Problem
We want to avoid fs-extra, and eventually move it to the banned import list. As a first step, we remove fs-extra with as much of our own `fs.ts` work, and `fs` when not possible. 

This PR only works on non-testing code. This way we can ensure our tests are not compromised, and still show that our code exhibits the desired behavior. 

### Indirect Translations
- Replace `ensureDir` with `mkDir` as suggested here: https://github.com/jprichardson/node-fs-extra/blob/HEAD/docs/ensureDir.md 
- Also `ensureDir` with permissions arg can be `mkDir` with `chmod` following. 
- Replace `move` with `rename` since we aren't using virtual file system, or crossing disk partitions: https://stackoverflow.com/questions/8579055/how-do-i-move-files-in-node-js

Otherwise the translations are fairly direct. 

## Notes
As of 9/30 there is a tech debt test failing blocking potential merge. This test was pushed back to 10/30. 

---

License: I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/shared/extensions/yaml.ts [hunk 1]
```diff
@@ -4,12 +4,12 @@
  */
 
 import * as vscode from 'vscode'
-import { readFileSync } from 'fs-extra'
 import { VSCODE_EXTENSION_ID } from '../extensions'
 import { getLogger } from '../logger/logger'
 import { getIdeProperties } from '../extensionUtilities'
 import { activateExtension } from '../utilities/vsCodeUtils'
 import { AWS_SCHEME } from '../constants'
+import * as fs2 from 'fs'
```
Comment (justinmk3) [inline packages/core/src/shared/extensions/yaml.ts:12]:
```suggestion
import * as nodefs from 'fs'
```

File: packages/core/src/shared/sam/debugger/goSamDebug.ts [hunk 2]
```diff
@@ -254,16 +254,16 @@ async function installDebugger(debuggerPath: string): Promise<boolean> {
...
         } else if (code) {
             getLogger().warn(`Install script did not sucessfully run, using old Delve binary...`)
         } else {
             getLogger().info(`Installed Delve debugger in ${debuggerPath}`)
         }
     } catch (e) {
-        if (installScript && (await fs2.exists(installScript.path))) {
-            fs.unlinkSync(installScript.path) // Removes the install script since it failed
+        if (installScript && (await fs.exists(installScript.path))) {
+            await unlink(installScript.path) // Removes the install script since it failed
```
Comment (justinmk3) [inline packages/core/src/shared/sam/debugger/goSamDebug.ts:266]: `fs.delete` ?

File: packages/core/src/shared/sam/localLambdaRunner.ts [hunk 3]
```diff
@@ -307,7 +307,7 @@ export async function runLambdaFunction(
         const payload =
             config.eventPayloadFile === undefined
                 ? undefined
-                : JSON.parse(await readFile(config.eventPayloadFile, { encoding: 'utf-8' }))
+                : JSON.parse(await fs.readFileAsString(config.eventPayloadFile))
```
Comment (justinmk3) [inline packages/core/src/shared/sam/localLambdaRunner.ts:310]: off-topic: we should rename this function :)
```suggestion
                : JSON.parse(await fs.readFileText(config.eventPayloadFile))
```

File: packages/core/src/shared/sshConfig.ts [hunk 4]
```diff
@@ -109,9 +110,17 @@ export class SshConfig {
         const sshConfigPath = getSshConfigPath()
         const section = this.createSSHConfigSection(proxyCommand)
         try {
-            await fs.ensureDir(path.dirname(path.dirname(sshConfigPath)), { mode: 0o755 })
-            await fs.ensureDir(path.dirname(sshConfigPath), 0o700)
-            await fs.appendFile(sshConfigPath, section, { mode: 0o600 })
+            const parentsDir = path.dirname(sshConfigPath)
+            const grandParentsDir = path.dirname(parentsDir)
+            // TODO: replace w/ fs.chmod once stub is merged.
```
Comment (justinmk3) [inline packages/core/src/shared/sshConfig.ts:115]: https://github.com/aws/aws-toolkit-vscode/pull/5647 is ready to merge

File: packages/core/src/shared/utilities/cliUtils.ts [hunk 5]
```diff
@@ -285,9 +285,7 @@ async function installSsmCli(
         await new TimedProcess('pkgutil', pkgArgs).run({ spawnOptions: { cwd: tempDir } })
         await new TimedProcess('tar', tarArgs).run({ spawnOptions: { cwd: tempPath } })
 
-        fs.copySync(path.join(tempPath, 'usr', 'local', 'sessionmanagerplugin'), outDir, {
-            recursive: true,
-        })
+        await fs.copy(path.join(tempPath, 'usr', 'local', 'sessionmanagerplugin'), outDir)
```
Comment (justinmk3) [inline packages/core/src/shared/utilities/cliUtils.ts:288]: `fs.copy` needs a docstring, it's not obvious that it also supports folders.

**Review by justinmk3 — APPROVED** [review]
LGTM!

File: packages/core/src/shared/utilities/cliUtils.ts (hunk 5 above)
Comment (Hweinstock) [reply packages/core/src/shared/utilities/cliUtils.ts:288]: tracked as a followup, seems like there a few other methods that could use them as well in `fs`.

**Comment by Hweinstock** [comment]
@aws/dexp @aws/codewhisperer-team

**Pull request merged**